### PR TITLE
[SR SQL Planner Summer Camp] New Parser and Analyzer support schema c…

### LINKED ISSUE #7941: New Parser and Analyzer support schema change (column change) operation-related statements（Challenging）
# Required work and output

1.  Use antlr4 to complete parse statements. The generated statement must follow the original data structure, and should be compatible with the old statement.
    
2.  In the original syntax file  **sql_parser.cup**, the new syntax needs to be compatible with the syntax of the old version, using the syntax file sql_parser.cup of the old version as the reference standard.
    
3.  Remove the logic of analytic functions in the statement and move the relevant logic to the new Analyzer framework. The new Analyzer is in the SQL/Analyzer directory.
    
4.  If the code related to Privilege Check exists in the original analyzer, move the code related to privilege control into PrivilegeChecker.
    
5.  When submitting the PR, please make sure there is unit test coverage for related changes.
    

# Tasks

 - [ ] AddColumnClause
 - [ ] AddColumnsClause
 - [ ] DropColumnClause
 - [ ] ModifyColumnClause
 - [ ] ShowColumnStmt
 - [ ] ColumnRenameClause
 - [ ] ReorderColumnsClause

# How to participate?

1.  Pick an issue you are interested in, click the issue link and leave a comment under the issue to sign up.  **One person can only choose one issue and the first person who picks the issue will be assigned.**
    
2.  **After the issue is assigned, you should complete the task in 14 days.** If you are unable to complete it within the specified time, the task will be withdrawn and assigned to others.
    
3.  Participants are required to produce two pieces of content (forms including but not limited to mind maps, study notes, blogs, and tutorials) in the course of the activity, each of which should be at least 200 words.
    
4.  Got questions? Join our  [Slack channel](https://join.slack.com/t/starrocks/shared_invite/zt-192zeqlc7-9MAvqAXNcveW_l5Pqr8r2g)  and find channel:  **#sql-planner**.
    

# Before you start

1.  Read  [Contributing.md](https://github.com/StarRocks/starrocks/blob/main/CONTRIBUTING.md)  carefully to understand how to submit a pull request and how to configure BE and FE development environment.
    
2.  Sign the  [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks)  before you make a pull request.
    
3.  To learn more about StarRocks, please refer to  [StarRocks documentation](https://github.com/StarRocks/docs/blob/main/TOC.md).
    

# Pull request requirements

1.  Create a pull request (PR) after you complete the task.
* PR title format: [SR SQL Planner Summer Camp] + PR title
2. After submitting PR, you can comment under the pinned Issue [SR SQL Planner Summer Camp].
* Comment format: "I have finished Issue #" + PR link

### PR DESCRIPTION
**Pull request by harveyyue**
…hange (column change) operation-related statements (#7941)

## What type of PR is this：
- [ ] bug
- [-] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7941

### CODE REVIEW

**Comment by kangkaisen** [comment]
@harveyyue Hi，please fix the failed FE UT, thanks.

File: fe/fe-core/src/main/java/com/starrocks/analysis/DropColumnClause.java [hunk 1]
```diff
@@ -67,7 +58,7 @@ public String toSql() {
         StringBuilder sb = new StringBuilder();
         sb.append("DROP COLUMN `").append(colName).append("`");
         if (rollupName != null) {
-            sb.append(" IN `").append(rollupName).append("`");
+            sb.append(" FROM `").append(rollupName).append("`");
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/analysis/DropColumnClause.java:61]: Why  need to change the `In` to `From`?  we should the behavior is consistent.

File: fe/fe-core/src/main/java/com/starrocks/analysis/AddColumnsClause.java [hunk 2]
```diff
@@ -39,7 +36,7 @@ public class AddColumnsClause extends AlterTableClause {
 
     private Map<String, String> properties;
     // set in analyze
-    private List<Column> columns;
+    private List<Column> columns = new ArrayList<>();
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/analysis/AddColumnsClause.java:39]: I think we could abstract a parent class for all these fix column classes, to reduce the duplicated Code

**Comment by harveyyue** [comment]
Guys, I open unit test url and get 404 error, please help to check.
http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/

File: fe/fe-core/src/main/java/com/starrocks/analysis/AddColumnsClause.java (hunk 2 above)
Comment (harveyyue) [reply fe/fe-core/src/main/java/com/starrocks/analysis/AddColumnsClause.java]: Add a abstract class AlterTableColumnClause to reduce duplicated code.

File: fe/fe-core/src/main/java/com/starrocks/analysis/DropColumnClause.java (hunk 1 above)
Comment (harveyyue) [reply fe/fe-core/src/main/java/com/starrocks/analysis/DropColumnClause.java]: Revert these change.

**Comment by kangkaisen** [comment]
> http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/
```
[com.starrocks.alter.AlterJobV2Test.testSchemaChange](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/AlterJobV2Test/testSchemaChange/)
[com.starrocks.alter.AlterJobV2Test.testModifyTableProperties](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/AlterJobV2Test/testModifyTableProperties/)
[com.starrocks.alter.AlterTest.testConflictAlterOperations](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/AlterTest/testConflictAlterOperations/)
[com.starrocks.alter.AlterTest.testCatalogModifyColumn](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/AlterTest/testCatalogModifyColumn/)
[com.starrocks.alter.AlterTest.testCatalogAddColumns](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/AlterTest/testCatalogAddColumns/)
[com.starrocks.alter.SchemaChangeJobV2Test.testSerializeOfSchemaChangeJob](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/SchemaChangeJobV2Test/testSerializeOfSchemaChangeJob/)
[com.starrocks.alter.SchemaChangeJobV2Test.testAddSchemaChange](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/SchemaChangeJobV2Test/testAddSchemaChange/)
[com.starrocks.alter.SchemaChangeJobV2Test.testModifyDynamicPartitionNormal](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/SchemaChangeJobV2Test/testModifyDynamicPartitionNormal/)
[com.starrocks.alter.SchemaChangeJobV2Test.testSchemaChangeWhileTabletNotStable](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/SchemaChangeJobV2Test/testSchemaChangeWhileTabletNotStable/)
[com.starrocks.alter.SchemaChangeJobV2Test.testModifyDynamicPartitionWithoutTableProperty](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/SchemaChangeJobV2Test/testModifyDynamicPartitionWithoutTableProperty/)
[com.starrocks.alter.SchemaChangeJobV2Test.testSchemaChange1](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.alter/SchemaChangeJobV2Test/testSchemaChange1/)
[com.starrocks.analysis.AlterTableStmtTest.testNormal](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.analysis/AlterTableStmtTest/testNormal/)
[com.starrocks.catalog.CreateTableTest.testAlterJsonTable](http://39.101.206.100:8090/job/starrocks_fe_unittest/27220/testReport/junit/com.starrocks.catalog/CreateTableTest/testAlterJsonTable/)
```

We will fix the 404 error ASAP. you could fix these UT first.

**Comment by harveyyue** [comment]
@kangkaisen, have fixed these ut test.

**Review by kangkaisen — DISMISSED** [review]

**Comment by kangkaisen** [comment]
> @kangkaisen, have fixed these ut test.

@harveyyue, Hi, still have one failed UT,  please fix it and rebase the latest main code, thanks.

```
com.starrocks.common.AnalysisException: Unknown table 'tbl1
	at com.starrocks.utframe.UtFrameUtils.parseStmtWithNewParser(UtFrameUtils.java:164)
	at com.starrocks.alter.AlterTest.testCatalogDropColumn(AlterTest.java:1873)
Caused by: com.starrocks.sql.analyzer.SemanticException: Unknown table 'tbl1
	at com.starrocks.sql.common.MetaUtils.getTable(MetaUtils.java:62)
	at com.starrocks.sql.analyzer.AlterTableStatementAnalyzer.analyze(AlterTableStatementAnalyzer.java:49)
	at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitAlterTableStatement(Analyzer.java:94)
	at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitAlterTableStatement(Analyzer.java:75)
	at com.starrocks.analysis.AlterTableStmt.accept(AlterTableStmt.java:129)
	at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:117)
	at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.analyze(Analyzer.java:77)
	at com.starrocks.sql.analyzer.Analyzer.analyze(Analyzer.java:72)
	at com.starrocks.utframe.UtFrameUtils.parseStmtWithNewParser(UtFrameUtils.java:158)
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 145 / 188 (77.13%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|6|00.00%|[534, 538, 542, 546, 550, 554]|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|0|1|00.00%|[62]|
|:large_blue_circle:|com/starrocks/analysis/ColumnRenameClause.java|1|2|50.00%|[67]|
|:large_blue_circle:|com/starrocks/analysis/ReorderColumnsClause.java|2|3|66.67%|[69]|
|:large_blue_circle:|com/starrocks/analysis/DropColumnClause.java|2|3|66.67%|[60]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterTableStatementAnalyzer.java|49|72|68.06%|[204, 208, 209, 216, 217, 222, 226, 240, 245, 246, 249, 263, 273, 277, 278, 285, 286, 299, 303, 308, 309, 318, 322]|
|:large_blue_circle:|com/starrocks/analysis/AddColumnClause.java|3|4|75.00%|[75]|
|:large_blue_circle:|com/starrocks/analysis/ModifyColumnClause.java|3|4|75.00%|[70]|
|:large_blue_circle:|com/starrocks/analysis/AddColumnsClause.java|3|4|75.00%|[74]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|67|74|90.54%|[516, 517, 518, 997, 998, 999, 1000]|
|:large_blue_circle:|com/starrocks/analysis/AlterTableColumnClause.java|15|15|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**